tests/test_algorithm/test_get_conditional_independencies.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+"""Test getting conditional independencies."""
+
+import unittest
+from typing import Set
+
+import y0.examples
+from y0.algorithm.conditional_independencies import are_d_separated, get_conditional_independencies
+from y0.examples import examples
+from y0.graph import NxMixedGraph
+from y0.struct import ConditionalIndependency
+
+
+class TestDSeparation(unittest.TestCase):
+    "Test the d-separation utility."
+
+    def test_mit_example(self):
+        graph = y0.examples.d_separation_example.graph.to_admg()
+
+        self.assertFalse(are_d_separated(graph, "AA", "B", given=["D", "F"]))
+        self.assertTrue(are_d_separated(graph, "AA", "B"))
+        self.assertTrue(are_d_separated(graph, "D", "E", given=["C"]))
+        self.assertFalse(are_d_separated(graph, "AA", "B", given=["C"]))
+        self.assertFalse(are_d_separated(graph, "D", "E"))
+        self.assertFalse(are_d_separated(graph, "D", "E", given=["AA", "B"]))
+        self.assertFalse(are_d_separated(graph, "G", "G", given=["C"]))
+
+
+class TestGetConditionalIndependencies(unittest.TestCase):
+    """Test getting conditional independencies."""
+
+    def assert_valid_ci_set(self, graph: NxMixedGraph, subject: Set[ConditionalIndependency]):
+        graph = graph.to_admg()
+        for ci in subject:
+            self.assertTrue(are_d_separated(graph, ci.left, ci.right,
+                                            given=ci.observations),
+                            f"Conditional independency is not a d-separation in {graph}")
+
+        gist = {(ci.left, ci.right) for ci in subject}
+        self.assertEqual(len(gist), len(set(gist)), "Duplicate left/right pair observed")
```

Comment:
since gist is a set this will always be equal. maybe define gist as a list comprehension instead?

Reply:
Ha, yes. Copy-paste error.  Fixed.